File: leetcode/2000/LC2090getAverages.go
func getAverages(nums []int, k int) []int {
	ret := make([]int, len(nums))
	for i := range ret {
		ret[i] = -1
	}

	win := k*2 + 1
	if len(nums) < win {
		return ret
	}

	left, right := 0, win
	var sum int64 = 0
	for i := 0; i < right; i++ {
		sum += int64(nums[i])
	}
	ret[right-k-1] = int(sum / int64(win))

	for right < len(nums) {
		sum += int64(nums[right])
		right++
		sum -= int64(nums[left])
		left++
		ret[right-k-1] = int(sum / int64(win))
	}

	return ret
}

File: leetcode/2000/LC2090GetAverages.java
import java.util.Arrays;

public class LC2090GetAverages {
    public int[] getAverages(int[] nums, int k) {
        int[] ret = new int[nums.length];
        Arrays.fill(ret, -1);

        int windowSize = 2 * k + 1;
        if (nums.length < windowSize) {
            return ret;
        }

        long sum = 0;
        int left = 0, right = windowSize;

        for (int i = 0; i < right; i++) {
            sum += nums[i];
        }
        ret[right - k-1] = (int)(sum / windowSize);

        while(right < nums.length){
            sum -= nums[left++];
            sum += nums[right++];
            ret[right - k-1] = (int)(sum / windowSize);
        }

        return ret;
    }
}
